File: 프로그래머스/1/12934. 정수 제곱근 판별/정수 제곱근 판별.cpp
#include<bits/stdc++.h>

using namespace std;

long long solution(long long n) 
{
    long long sqrt_n = sqrt(n);
    
    if (sqrt_n * sqrt_n == n) 
    {
        return (sqrt_n + 1) * (sqrt_n + 1);
    }
    return -1;
}

File: 프로그래머스/1/12934. 정수 제곱근 판별/정수 제곱근 판별.cs
using System;

public class Solution {
    public long solution(long n) {

        double sqrt = Math.Sqrt(n);
        
        if (sqrt % 1 == 0) 
        {
            long x = (long)sqrt;
            return (x + 1) * (x + 1);
        } 
        else return -1;
        
    }
}